integrationtests/src/test/java/io/kroxylicious/net/MultiTenantSniInetAddressResolverProvider.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.net;
+
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.net.spi.InetAddressResolver;
+import java.net.spi.InetAddressResolverProvider;
+import java.util.stream.Stream;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class MultiTenantSniInetAddressResolverProvider extends InetAddressResolverProvider {
```

Comment:
I'm quite tempted to say we should extract this from the POC and include it in kroxy regardless of what happens to this POC.

Reply:
As per the comments above, MultiTenantSniInetAddressResolverProvider was just the simplest thing that worked to get the POC running.